### PR TITLE
Fastsum mex

### PR DESCRIPTION
**Pull request by michaelquellmalz**
- added mex interface for fastsum
- split init and precomputation in fastsum for source and target nodes
- added iNFFT_1d based on fastsummex
- added mex inteface for nfsoft
- added choice of ovesampling factor no nfsoft
- small fix for doxygen

### CODE REVIEW

File: applications/fastsum/fastsum.c [hunk 1]
```diff
@@ -582,22 +590,22 @@ static C SearchBox(R *y, fastsum_plan *ths)
   }
   else
   {
-    exit(EXIT_FAILURE);
+    return 0.0/0.0; //exit(EXIT_FAILURE);
```
Comment (jenskeiner) [inline applications/fastsum/fastsum.c]: There does not seem to be a portable way to define a NaN. Even C99 doesn't require systems to support them. Can't we just return zero and document that the result is undefined for dimensions larger than 3?

File: applications/fastsum/fastsum.c [hunk 2]
```diff
@@ -939,37 +1106,10 @@ void fastsum_precompute(fastsum_plan *ths)
...
-#endif
-
-#ifdef MEASURE_TIME
-  t0 = getticks();
-#endif
   /** init NFFT plan for transposed transform in first step*/
-  for (k = 0; k < ths->mv1.M_total; k++)
-    for (t = 0; t < ths->mv1.d; t++)
-      ths->mv1.x[ths->mv1.d * k + t] = -ths->x[ths->mv1.d * k + t]; /* note the factor -1 for transposed transform instead of adjoint*/
+//  for (k = 0; k < ths->mv1.M_total; k++)
```
Comment (jenskeiner) [inline applications/fastsum/fastsum.c]: You may want to remove the stuff you have commented out.

**Comment by michaelquellmalz** [comment]
Is there a reason the merge was squashed to one commit? That way the contributions of others like @melaniekircheis do not show up in the history of develop anymore.

**Comment by jenskeiner** [comment]
Should re-open this one to merge without squashing.

**Pull request merged**